packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3161,14 +3165,29 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
 
     final bool wasSemanticsBoundary = _semantics != null && (_cachedSemanticsConfiguration?.isSemanticBoundary ?? false);
     _cachedSemanticsConfiguration = null;
+
     bool isEffectiveSemanticsBoundary = _semanticsConfiguration.isSemanticBoundary && wasSemanticsBoundary;
+
+    bool mayProduceSiblingNodes =
+      _cachedSemanticsConfiguration?.childConfigurationsDelegate != null ||
```

Comment:
Isn't _cachedSemanticsConfiguration always null here? we just set it to null two lines above?

Reply:
you are right, I should move this line up before reset it